biojava-structure/src/main/java/org/biojava/nbio/structure/io/cif/CifStructureConsumerImpl.java

```diff
@@ -1297,18 +1297,22 @@ public void finish() {
             SeqMisMatch seqMisMatch = new SeqMisMatchImpl();
             seqMisMatch.setDetails(structRefSeqDif.getDetails().get(rowIndex));
 
-            String insCode = structRefSeqDif.getPdbxPdbInsCode().get(rowIndex);
+            String insCode = null;
+            if (structRefSeqDif.getPdbxPdbInsCode().isDefined()) {
+                insCode = structRefSeqDif.getPdbxPdbInsCode().get(rowIndex);
                 if ("?".equals(insCode)) {
-                insCode = null;
+                    insCode = null;
+                }
             }
             seqMisMatch.setInsCode(insCode);
             seqMisMatch.setOrigGroup(structRefSeqDif.getDbMonId().get(rowIndex));
             seqMisMatch.setPdbGroup(structRefSeqDif.getMonId().get(rowIndex));
-            seqMisMatch.setPdbResNum(structRefSeqDif.getPdbxAuthSeqNum().get(rowIndex));
-            seqMisMatch.setUniProtId(structRefSeqDif.getPdbxSeqDbAccessionCode().get(rowIndex));
+            seqMisMatch.setPdbResNum(structRefSeqDif.getPdbxAuthSeqNum().isDefined()? structRefSeqDif.getPdbxAuthSeqNum().get(rowIndex):null);
+            seqMisMatch.setUniProtId(structRefSeqDif.getPdbxSeqDbAccessionCode().isDefined()? structRefSeqDif.getPdbxSeqDbAccessionCode().get(rowIndex):null);
             seqMisMatch.setSeqNum(structRefSeqDif.getSeqNum().get(rowIndex));
 
-            String strandId = structRefSeqDif.getPdbxPdbStrandId().get(rowIndex);
+            String strandId = structRefSeqDif.getPdbxPdbStrandId().isDefined()? structRefSeqDif.getPdbxPdbStrandId().get(rowIndex) : null;
+            if (strandId == null) continue;
```

Comment:
```suggestion
            if (!structRefSeqDif.getPdbxPdbStrandId().isDefined()) continue;
            String strandId = structRefSeqDif.getPdbxPdbStrandId().get(rowIndex);
```

for fewer logic/assignments.